test/conformance/walk.go

```diff
@@ -74,39 +83,43 @@ type conformanceData struct {
 	Description string
 }
 
-func (v *visitor) convertToConformanceData(at *ast.BasicLit) {
+func (v *visitor) convertToConformanceData(n ast.Node, tName string) {
 	cd := conformanceData{}
-
-	comment := v.comment(at)
-	pos := v.FileSet.Position(at.Pos())
-	cd.URL = fmt.Sprintf("%s%s#L%d", *baseURL, pos.Filename, pos.Line)
-
-	lines := strings.Split(comment, "\n")
-	cd.Description = ""
-	for _, line := range lines {
-		line = strings.TrimSpace(line)
-		if strings.HasPrefix(line, "Testname:") {
-			line = strings.TrimSpace(line[9:])
-			cd.TestName = line
-			continue
-		}
-		if strings.HasPrefix(line, "Description:") {
-			line = strings.TrimSpace(line[12:])
+	switch at := n.(type) {
+	case *ast.BasicLit, *ast.CallExpr, *ast.BinaryExpr:
+		comment := v.comment(at)
+		pos := v.FileSet.Position(at.Pos())
+		cd.URL = fmt.Sprintf("%s%s#L%d", *baseURL, pos.Filename, pos.Line)
+
+		lines := strings.Split(comment, "\n")
+		cd.Description = ""
+		for _, line := range lines {
+			line = strings.TrimSpace(line)
+			if strings.HasPrefix(line, "Testname:") {
```

Comment:
Seems like "Testname:" and "Description:" should be const. Then we can do things like call len() on this for things like indexing and the code is both more clear and easier to maintain.

Reply:
Done.